### PR TITLE
Move documentation to the docs directory

### PR DESCRIPTION
**Pull request by pellared**
## Why

> Why this pull request was created? Explain the problem that this pull request is trying to solve.

Avoid clutter in the "frontpage" of the repository

Reference: https://github.com/goyek/goyek/discussions/58#discussioncomment-782455

## What

> What does this pull request contain? Explain what is this pull request changing.

- Move documentation to the `docs` directory. See: https://github.com/pellared/goyek/tree/docs
- Pin versions in build GH workflow (could be done as a separate PR in theory)

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated.
- [x] The code changes follow [Effective Go](https://golang.org/doc/effective_go).

### CODE REVIEW

**Comment by dertseha** [comment]
OK, thank you for the initiative - though the current changes are a bit too harsh for the start.

I didn't mean that *no* file should be available -- a `README.md` still should be provided in the root folder, as this file is (probably) expected by (at least) GitHub and GoDoc. It is the first "what is this?" information. Or, said differently; I'd look for this file in the root folder of the repository and not start searching anywhere else.
~~I believe this is similar for at least the CoC file as well - GitHub expects this to be in the root.~~ Though, GitHub also looks in the `docs` subfolder at least for CoC, as [described here](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project#adding-a-code-of-conduct-manually).

Within the `docs` folder, we can let go of the SCREAMING filenames. The uppercase format is only a remnant of the old days (see [here](https://en.wikipedia.org/wiki/README), and only "common" in the root folder and for specific files. Within the folder for documentation I expect everything equally informative - so in that folder I propose the `lowercase-spear-format.md`.

In short: root folder can keep the "common" screaming files, with readme having the main selling points, and `docs` has detail information for further interests.

**Comment by pellared** [comment]
What MD files do you want in root apart from `README.md` ?

**Comment by pellared** [comment]
@sheldonhull PTAL

**Review by dertseha — APPROVED** [review]

**Comment by dertseha** [comment]
readme.md in the root folder is OK, this is common.
